lib/qa_topic_edits.rb

```diff
@@ -61,10 +61,32 @@ class ::Topic
 
   def self.voted(topic, user)
     return nil if !user || !SiteSetting.qa_enabled
+    return nil if !self.can_vote(topic, user)
+  end
+
+  def self.can_vote(topic, user)
+    return nil if !user || !SiteSetting.qa_enabled
+    vote_count = self.vote_count(topic, user)
+
+    case user.trust_level
```

Comment:
We can re-write this case statement as a single line using string interpolation and the "send" method:
```
vote_limit = SiteSetting.send("qa_tl#{user.trust_level}_vote_limit")
```
You can do any string interpolation in ruby using #{variable} insde " " (but not inside ' ').

There's a good explaination of what ``send`` does [here](https://stackoverflow.com/questions/3337285/what-does-send-do-in-ruby).

Reply:
Oh that's so much cleaner! And interesting, it's like a top down filter sort of, that goes down the variables until it hits a match?  Anyhoo, this is updated.